R/utils_render_latex.R

```diff
@@ -314,3 +332,181 @@ create_source_note_component_l <- function(data) {
 
   source_note_component
 }
+
+# Function to build a vector of `group` rows in the table body
+create_group_rows <- function(n_rows,
+                              groups_rows_df) {
+
+  unname(
+    unlist(
+      lapply(
+        seq(n_rows),
+        FUN = function(x) {
+
+          if (!(x %in% groups_rows_df$row_start)) {
+            return("")
+          }
+
+          latex_group_row(
+            group_name = groups_rows_df[
+              which(groups_rows_df$row_start %in% x), "group_label"][[1]],
+            top_border = x != 1,
+            bottom_border = x != n_rows
+          )
+        }
+      )
+    )
+  )
+}
+
+# Function to build a vector of `body` rows
+create_body_rows <- function(n_rows,
+                             row_splits_body) {
+
+  unname(
+    unlist(
+      lapply(
+        seq_len(n_rows),
+        FUN = function(x) {
+          latex_body_row(content = row_splits_body[[x]], type = "row")
+        }
+      )
+    )
+  )
+}
+
+# Function to build a vector of `summary` rows in the table body
+create_summary_rows <- function(n_rows,
+                                n_cols,
+                                boxh,
+                                list_of_summaries,
+                                groups_rows_df,
+                                stub_available,
+                                summaries_present) {
+
+  default_vars <- dplyr::pull(dplyr::filter(boxh, type == "default"), "var")
+
+  unname(
+    unlist(
+      lapply(
+        seq(n_rows),
+        FUN = function(x) {
+
+          if (!stub_available ||
+              !summaries_present ||
+              !(x %in% groups_rows_df$row_end)) {
+            return("")
+          }
+
+          group <-
+            groups_rows_df %>%
+            dplyr::filter(row_end == x) %>%
+            dplyr::pull(group_id)
+
+          if (!(group %in% names(list_of_summaries$summary_df_display_list))) {
+            return("")
```

Comment:
Check if this is supposed to even be possible (if not, maybe error?).

Reply:
This is not what I thought it was during the review. The logic is to check whether this group has a corresponding entry in `list_of_summaries$summary_df_display_list` (i.e., are there summary rows for this group?); if not, return an empty string. This is not for checking whether the named table for summary rows actually has rows.